app/services/savings_pass_fail_service.rb

```diff
@@ -0,0 +1,67 @@
+class SavingsPassFailService
+
+  def initialize(saving)
+    @saving = saving
+  end
+
+  def calculate!
+    @saving.fee_threshold = FeeThreshold.new(@saving.application.fee).band
+    @saving.passed = calculate_pass_fail
+    unless @saving.passed
```

Comment:
I would maybe abstract this to a method, to make the `calculate!` more readable. Something like `set_application_outcome_when_failed` or so.